episodes/continuous-integration.md

```diff
@@ -470,3 +480,11 @@ A small graph is shown and at the bottom "release-dists" link is provided.
 Click on that to download the package.
 
 That's it! Package published!
+
+## Considerations
+
+CI/CD pipelines are not very suitable if your tests requires a lot of static data.
+Running large integration tests inside a CI/CD pipeline is thus not recommended as there is generally limited space and time in CI/CD pipelines.
+Writing small and fast unit tests that run automatically inside CI/CD pipelines rather than large integration tests is recommended.
```

Comment:
I see the point of limiting CI/CD usage depending on available resources, but one of the benefits I see of CI/CD is exactly for running longer integration tests automatically: If I have some longer integration tests that might take a few minutes, I don't want to locally run my test suite each time I make a commit. Having those still run by CI/CD makes sure they're still used, even if I don't frequently/ever run them locally.

I would probably set it up such that it might only run e.g. before a release though, which I can see is a more advanced use case. Do you think there is space to be more (explicitly) nuanced about this? Or would it only cause confusion?

Reply:
Ik heb wat nuance aangebrach in 5105cc6.

Reply:
For users who run into this type of issues, maybe it is worth pointing that there are other alternatives, like self-hosted runners.

Here are some links which we could point people to:
https://github.com/ci-for-research/self-hosted-runners https://tilburgsciencehub.com/topics/automation/automation-tools/deployment/ghactions-self-hosted-runner/

I imagine 90% of people who are just getting started won't need this, but maybe good for them to know that if they ever run into issues of tests taking too long, there are alternatives.

Reply:
Good point. Added in commit 5880255